app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarActivity.java

```diff
@@ -438,9 +438,16 @@ public void onEditEvent(@NotNull CalendarItem calendarItem) {
 
     @Override
     public void onEventClick(WeekViewEvent weekViewEvent, RectF rectF) {
-        CalendarItem item = calendarController.getCalendarItemByStartAndEndTime(
-                new DateTime(weekViewEvent.getStartTime()),
-                new DateTime(weekViewEvent.getEndTime()));
+        String eventId = weekViewEvent.getIdentifier();
+        if (eventId == null) {
+            return;
+        }
+
+        CalendarItem item = calendarController.getCalendarItemById(eventId);
+        if (item == null) {
+            return;
```

Comment:
This shoudn't happen, if everything was set up correctly.
Could we add some logging here, so we maybe get a chance to debug this eventually?